### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [f-prime/fist#14]
Would you like to add more error handling for return values from functions like the following?
- [malloc](https://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html "Memory allocation") ⇒ [dappendc](https://github.com/f-prime/fist/blob/99a9710a3d4f8e0e6e7a090925027ce804e37f65/fist/dstring.c#L10)
- [printf](https://pubs.opengroup.org/onlinepubs/9699919799/functions/printf.html "Print formatted output") ⇒ [handle_connection](https://github.com/f-prime/fist/blob/69acb7457007fddee3501f2e47704d0faa94d3c5/fist/server.c#L28)

**Comment by f-prime** [comment]
Absolutely. Is that something you want to work on, or is this a suggestion?

**Comment by elfring** [comment]
I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Checking more return values") a bit more.
Would you like to detect every error situation as early as possible?

**Comment by f-prime** [comment]
@elfring Ideally yes.

**Comment by elfring** [comment]
How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2C_C.2B.2B "Tools for code analysis") also for this software?

-----

* Are you interested to apply [aspect-oriented software development](https://aspectc.org/ "Software development with AspectC++?")?
* Would you like to encapsulate error detection and corresponding exception handling as [a reusable aspect](https://en.wikipedia.org/wiki/Aspect_%28computer_programming%29 "Combination of a pointcut with advice") in this software?

**Comment by f-prime** [comment]
I don't know much about this but I would be open to hear more about it if you'd be willing to share why you think it is useful.

**Comment by elfring** [comment]
I suggest to read a bit more advice from linked information sources.
